registry/registry.go

```diff
@@ -133,9 +114,11 @@ func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
 // DockerHeaders returns request modifiers that ensure requests have
 // the User-Agent header set to dockerUserAgent and that metaHeaders
 // are added.
-func DockerHeaders(metaHeaders http.Header) []transport.RequestModifier {
+func DockerHeaders(userAgent string, metaHeaders http.Header) []transport.RequestModifier {
 	modifiers := []transport.RequestModifier{
-		transport.NewHeaderRequestModifier(http.Header{"User-Agent": []string{dockerUserAgent}}),
+		transport.NewHeaderRequestModifier(http.Header{
+			"User-Agent": []string{userAgent},
```

Comment:
Based on the signature of this function, an error would be surprising. If this function was expected to enforce that the user agent has a proper Docker versioned user agent string, I would expect the signature to take in docker version information rather than a user agent string and construct the user agent. I do not feel strongly about this though, I think the given signature is straight forward and better without returning an error. The comment should be updated to match the updated behavior though.